framework/src/actions/CreateDisplacedProblemAction.C

```diff
@@ -26,23 +26,35 @@ InputParameters validParams<InitDisplacedProblemAction>()
   return params;
 }
 
-InitDisplacedProblemAction::InitDisplacedProblemAction(InputParameters parameters) :
+CreateDisplacedProblemAction::CreateDisplacedProblemAction(InputParameters parameters) :
     Action(parameters)
 {
 }
 
-InitDisplacedProblemAction::~InitDisplacedProblemAction()
+CreateDisplacedProblemAction::~CreateDisplacedProblemAction()
 {
 }
 
 void
-InitDisplacedProblemAction::act()
+CreateDisplacedProblemAction::act()
 {
-  if (isParamValid("displacements"))
+  if (isParamValid("displacements") && _displaced_mesh.get() != NULL)
```

Comment:
What would cause the `_displaced_mesh` to by NULL at this point if `displacements` are on?  Should this be checked for as an error condition instead?